src/sql/src/catalog.rs

```diff
@@ -1370,6 +1370,61 @@ impl RustType<proto::ObjectType> for ObjectType {
     }
 }
 
+/// Specification for objects that will be affected by a default privilege.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct DefaultPrivilegeObject {
+    /// The role id that created the object.
+    pub role_id: RoleId,
+    /// The database that the object is created in if Some, otherwise all databases.
+    pub database_id: Option<DatabaseId>,
+    /// The schema that the object is created in if Some, otherwise all databases.
+    pub schema_id: Option<SchemaId>,
+    /// The type of object.
+    pub object_type: ObjectType,
+}
+
+impl DefaultPrivilegeObject {
+    /// Creates a new [`DefaultPrivilegeObject`].
+    pub fn new(
+        role_id: RoleId,
+        database_id: Option<DatabaseId>,
+        schema_id: Option<SchemaId>,
+        object_type: ObjectType,
+    ) -> DefaultPrivilegeObject {
+        DefaultPrivilegeObject {
+            role_id,
+            database_id,
+            schema_id,
+            object_type,
+        }
+    }
+}
+
+/// Specification for the privileges that will be granted from default privileges.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct DefaultPrivilegeAclItem {
+    /// The role that will receive the privileges.
+    pub grantee: RoleId,
+    /// The specific privileges granted.
+    pub acl_mode: AclMode,
+}
+
+impl DefaultPrivilegeAclItem {
+    /// Creates a new [`DefaultPrivilegeAclItem`].
+    pub fn new(grantee: RoleId, acl_mode: AclMode) -> DefaultPrivilegeAclItem {
+        DefaultPrivilegeAclItem { grantee, acl_mode }
+    }
+
+    /// Converts this [`DefaultPrivilegeAclItem`] into an [`MzAclItem`].
+    pub fn mz_acl_item(self, grantor: RoleId) -> MzAclItem {
+        MzAclItem {
+            grantee: self.grantee,
+            grantor,
+            acl_mode: self.acl_mode,
+        }
+    }
+}
+
```

Comment:
This was moved from `src/adapter/src/catalog.rs`, but not changed, to make the structs more accessible.